### DISCUSSION TITLE
What's the best practice to use dbmate in k8s?

### DISCUSSION THREAD

**Discussion by simsicon** [amacneil/dbmate#238] [Support] [answered]
How do you use dbmate in k8s environment?

**Comment by amacneil** [comment, marked as answer]
I haven't personally tried this, but know folks who have. You can probably run it as a Job pod?

How to trigger is up to you.

**Reply by simsicon** [reply]
Thanks, I think that's it, a job pod will do that.

**Comment by pawndev** [comment]
Personnaly, I'm using it in an [`initContainer`](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/).
You can take your application docker image and run the migration from it. Or, you can put all your migrations in a PersistentVolume/ConfigMap, and mount it in you containers/initContainers.

That is the way I'm using it.

**Comment by gregwebs** [comment]
`initContainer` is good for auto-migrating. Another approach is to send a notification when there are unapplied migrations.

Dockerfile that adds the migrations into the image.

``` Dockerfile
FROM amacneil/dbmate:2.2
RUN apk add curl

ADD . /sql/dbmate
WORKDIR /sql/dbmate
ENTRYPOINT ["/bin/sh", "./pending-check.sh"]
```

notification script

``` sh
#!/usr/bin/env bash
set -euo pipefail
export DATABASE_URL=postgres://${PGUSER}:${PGPASSWORD}@${PGHOST}/${PGDATABASE}
dbmate status
if dbmate status | grep Pending >/dev/null ; then
    unapplied=$(dbmate status | grep '\[ \]')
    # notify about unapplied migrations here
fi
```
